spec/API_specification/linear_algebra_functions.md

```diff
@@ -275,9 +275,35 @@ TODO
 TODO
 
 (function-svd)=
-### svd()
+### svd(x, /, *, compute_uv=True, full_matrices=True)
 
-TODO
+Computes the singular value decomposition `A = USV` of a matrix (or stack of matrices) `x`.
+
+#### Parameters
+
+-   **x**: _&lt;array&gt;_
+
+    -   input array having shape `(..., M, N)` and whose innermost two dimensions form matrices on which to perform singular value decomposition. Must have a data type of either `float32` or `float64`.
+
+-   **compute_uv**: _bool_
+
+    -   If `True`, compute the left and right singular vectors and return as `u` and `v`, respectively. Default: `True`.
+
+-   **full_matrices**: _bool_
+
+    -   If `True`, compute full-sized `u` and `v`, such that `u` has shape `(..., M, M)` and `v` has shape `(..., N, N)`. If `False`, compute on the leading `K` singular vectors, such that `u` has shape `(..., M, K)` and `v` has shape `(..., K, N)` and where `K = min(M, N)`. This option must be ignored if `compute_uv` is `False`. Default: `True`.
+
+#### Returns
+
+-   **out**: _Union\[ &lt;array&gt;, Tuple\[ &lt;array&gt;, ... ] ]_
+
+    -   if `compute_uv` is `False`, an array with shape `(..., K)` that contains the vector(s) of singular values of length `K`. For each vector, the singular values must be sorted in descending order by magnitude, such that `s[..., 0]` is the largest value, `s[..., 1]` is the second largest value, et cetera. The first `x.ndim-2` dimensions must have the same shape as those of the input `x`.
```

Comment:
Perhaps to be consistent across the `linalg` spectrum (ref: #95), this should return a namedtuple `(s,)` too?

Reply:
I think those are all mistakes. A tuple with a single array in it is kind of weird and unpythonic. Doesn't matter if it's a namedtuple or not.

The design consistency is quite hard to see across all these linalg PRs. We need the overview PR with design principles for all linalg functions first I think.

Reply:
Sounds good. Yeah I left a bunch of questions yesterday about whether it should be a namedtuple or just an array when there is only one object to be returned.

Reply:
This should now be resolved due to splitting of `svd` into `svd` and `svdvals`.